Fix #3931: Add `unversioned` flag to `Package.set()` and `Package.set_dir()` (#3927)
quiltdata/quilt: docs/CHANGELOG.md

# unreleased - YYYY-MM-DD
## Python API

> * [Added] New 'unversioned' parameter to `set_dir` for use with HealthOmics and other S3 URIs that lack permissions for `list_object_versions` ([#3927])

Reviewer: ```suggestion
* [Added] New 'unversioned' parameter to `Package.set_dir()` and `Package.set()` for use with HealthOmics and other S3 URIs that lack permissions for `ListBucketVersions` and/or `GetObjectVersion` ([#3927](https://github.com/quiltdata/quilt/pull/3927))
```
I'm not sure that `S3 URIs that lack permissions` is semantically correct

> * [Added] New 'unversioned' parameter to `Package.set_dir()` and `Package.set()` for use with HealthOmics and other S3 URIs that lack permissions for `ListBucketVersions` and/or `GetObjectVersion` ([#3927](https://github.com/quiltdata/quilt/pull/3927))

Reviewer: I'm not sure that S3 URIs that lack permissions is semantically correct

Author: Any suggestions? The term "buckets that lack sufficient permissions" feels even less accurate, since Omics has S3 URIs but no actual "buckets" :-(

Reviewer: I mean that it's user lacks permissions to "S3 URIs"

> * [Added] New 'unversioned' parameter to `Package.set_dir()` and `Package.set()` for use with HealthOmics and other S3 URIs that lack permissions for `ListBucketVersions` and/or `GetObjectVersion` ([#3927](https://github.com/quiltdata/quilt/pull/3927))

Reviewer 2: ```suggestion
* [Added] New 'unversioned' parameter to `Package.set_dir()` and `Package.set()` for use with S3 URIs, such as HealthOmics, that do not support `ListBucketVersions` and/or `GetObjectVersion` ([#3927](https://github.com/quiltdata/quilt/pull/3927))
```
